### DISCUSSION TITLE
Access JWT Secret and Other Secrets Programmatically?

### DISCUSSION THREAD

**Discussion by Dark-Akkula** [supabase/supabase#20600] [Questions] [answered]
I would like to use the pgjwt extension to mint some JWTs for Supabase login.  To do this, I would need to use the JWT Secret that is hidden on the API page.  Is there way I can access this value programmatically using a select statement in a procedure or something without having to duplicate it somewhere in the vault or other database?  If so, where is it located?

Otherwise,  I would have to copy and paste it into the vault which isn't really ideal to have it in two places to inevitably become out of sync.

**Comment by GaryAustin1** [comment, marked as answer]
It is not available thru a database call at this time but is supposedly being worked on to allow access thru database settings.  Or at least I've not seen it announced or in my instances yet.

**Reply by Dark-Akkula** [reply]
Okay, I guess I will have to put the secret in the vault for now.  Thanks!
